### PR TITLE
Configure PR checks to run on Windows and MacOS

### PR DESCRIPTION
**Pull request by JuliaABurch**
## Description
To avoid issues in the future where code changes prevent the repo from compiling on Windows, I've added it as part of the configuration for the PR-check Github action; this action will be run for each OS specified under `strategy`. Since the majority of the developers use MacOSX, I've added that as well. These jobs run in parallel, so the time taken to run PR checks should not increase by any significant amount.

I've removed the `chromium --version` check as the command is not cross-platform compatible, the version of chrome on the path may not be the chrome the karma-runner is using, and trying to run the right command would add unneeded complexity to the file. Chrome is installed on all OS images by default, and the karma-runner prints out what version it's running with.

You can see an example of the PR action being run on all platforms here: https://github.com/JuliaABurch/amazon-sumerian-hosts/pull/3 (Note that the windows run will fail until  #109 is merged, but I have it passing locally as I pulled in that change to my repo )

## Reviewer Testing Instructions
Github actions are a little tricky to test - I did it on my forked repo after manually enabling actions.

## Submission Checklist
I confirm that I have...
- [x] removed hard-coded Cognito IDs
- [x] manually smoke-tested the BabylonJS integration tests
- [x] manually smoke-tested the BabylonJS demos
- [x] manually smoke-tested the Three.js integration tests
- [x] manually smoke-tested the Three.js demo

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -9,10 +9,13 @@ on:
...
   TEST_BROWSERS: ChromeHeadless
-  
+
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (eherozhao) [inline .github/workflows/pull-request.yml:18]: will this matrix cover all possibilities?

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -22,8 +25,7 @@ jobs:
           node-version: '16.x'
           registry-url: 'https://registry.npmjs.org'
       - run: npm install
-      - run: chromium --version
```
Comment (eherozhao) [inline .github/workflows/pull-request.yml:25]: q: why do we remove this line?

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -22,8 +26,7 @@ jobs:
...
       - run: npm run build-test
       - name: Perform CodeQL Analysis
+        if: matrix.os == 'ubuntu-latest' # we only need to run this check on one platform
```
Comment (eherozhao) [inline .github/workflows/pull-request.yml:31]: nit: any indentation needed below this line?

Comment (eherozhao) [reply .github/workflows/pull-request.yml:31]: nit: do we need a hyphen before `if`?

**Review by eherozhao — COMMENTED** [review]
How do we test the GitHub Action?

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (JuliaABurch) [reply .github/workflows/pull-request.yml:18]: What possibilities am I missing?

Comment (eherozhao) [reply .github/workflows/pull-request.yml:18]: I am just not sure whether `-lastest` is able to cover all OS possibilities. Or we may don't have to cover all?

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (JuliaABurch) [reply .github/workflows/pull-request.yml:31]: > How do we test the GitHub Action?

You can actually see it running in this PR - if you scroll down, you'll see the tests being run on the three platforms specified. 

I can update the PR description if it's not very clear as to how I did it, but I used my forked repository to test after enabling github actions there.

**Comment by JuliaABurch** [comment]
> How do we test the GitHub Action?

You can actually see it running in this PR - if you scroll down, you'll see the tests being run on the three platforms specified. 

I can update the PR description if it's not very clear as to how I did it, but I used my forked repository to test after enabling github actions there.

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (JuliaABurch) [reply .github/workflows/pull-request.yml:31]: A hyphen indicates a command or a step to execute, so we don't want a hyphen on `if` as we want to conditionally run the command it applies to.

Also, the indentation looks fine to me, but let me know if I'm missing something

**Comment by eherozhao** [comment]
> > How do we test the GitHub Action?
> 
> You can actually see it running in this PR - if you scroll down, you'll see the tests being run on the three platforms specified.
> 
> I can update the PR description if it's not very clear as to how I did it, but I used my forked repository to test after enabling github actions there.

I see. That looks pretty cool! Thank you!

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (eherozhao) [reply .github/workflows/pull-request.yml:31]: we don't need `use` to be a part of the `if` block?

**Review by eherozhao — APPROVED** [review]

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (JuliaABurch) [reply .github/workflows/pull-request.yml:31]: The `if` applies to the whole block, which is scoped by the hyphen

**Review by prestomation — APPROVED** [review]

**Pull request merged**